### PR TITLE
REST > Billing getOne/getMany/Download/Webhook doc + schemas

### PR DESCRIPTION
**Pull request by Ninjeneer**
\+ postman collection update

### CODE REVIEW

File: src/assets/server/rest/v1/schemas/billing/invoices-get.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  "$id": "invoices-get",
+  "title": "Get invoices list Request",
+  "type": "object",
+  "properties": {
+    "UserID": {
+      "$ref": "common#/definitions/ids"
+    },
+    "Status": {
+      "type": "string",
+      "pattern": "^((paid|open|draft|void|uncollectible|deleted)(\\||paid|open|draft|void|uncollectible|deleted)*)?$",
```
Comment (ClaudeROSSI) [inline src/assets/server/rest/v1/schemas/billing/invoices-get.json:11]: The regex seems wrong to me!
Please have a look to my suggestion below:

```suggestion
      "pattern": "^((paid|open|draft|void|uncollectible|deleted)(\|paid|\|open|\|draft|\|void|\|uncollectible|\|deleted)*)?$",
```

File: src/server/rest/v1/service/BillingService.ts [hunk 2]
```diff
@@ -347,7 +347,7 @@ export default class BillingService {
     UtilsService.assertComponentIsActiveFromToken(req.user, TenantComponents.BILLING,
       Action.LIST, Entity.INVOICE, MODULE_NAME, 'handleGetInvoice');
     // Filter
-    const filteredRequest = BillingSecurity.filterGetInvoiceRequest(req.query);
+    const filteredRequest = BillingValidator.getInstance().validateBillingInvoicesGetByIdReq(req.query);
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/service/BillingService.ts:350]: Name is not similar to the one used by other validators

See my proposal:
```suggestion
    const filteredRequest = BillingValidator.getInstance().validateBillingInvoiceGetReq(req.query);
```

File: src/server/rest/v1/service/BillingService.ts [hunk 3]
```diff
@@ -618,7 +618,7 @@ export default class BillingService {
     UtilsService.assertComponentIsActiveFromToken(req.user, TenantComponents.BILLING,
       Action.DOWNLOAD, Entity.BILLING, MODULE_NAME, 'handleDownloadInvoice');
     // Filter
-    const filteredRequest = BillingSecurity.filterDownloadInvoiceRequest(req.query);
+    const filteredRequest = BillingValidator.getInstance().validateBillingInvoicesGetByIdReq(req.query);
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/service/BillingService.ts:621]: idem. Name is confusing:

```suggestion
    const filteredRequest = BillingValidator.getInstance().validateBillingInvoiceGetReq(req.query);
```

File: src/server/rest/v1/service/security/BillingSecurity.ts [hunk 4]
```diff
@@ -65,7 +65,7 @@ export default class BillingSecurity {
 
   public static filterBillingWebHookRequest(requestQuery: any): HttpBillingWebHookRequest {
     return {
-      tenantID: sanitize(requestQuery.TenantID)
+      TenantID: sanitize(requestQuery.TenantID)
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/service/security/BillingSecurity.ts:68]: Method is not used anymore.
Please remove it.

File: src/server/rest/v1/validator/BillingValidator.ts [hunk 5]
```diff
@@ -12,6 +12,9 @@ export default class BillingValidator extends SchemaValidator {
   private billingGetUserPaymentMethods: Schema = JSON.parse(fs.readFileSync(`${global.appRoot}/assets/server/rest/v1/schemas/billing/billing-payment-methods-get.json`, 'utf8'));
   private billingDeleteUserPaymentMethod: Schema = JSON.parse(fs.readFileSync(`${global.appRoot}/assets/server/rest/v1/schemas/billing/billing-delete-payment-method.json`, 'utf8'));
   private billingSetupUserPaymentMethod: Schema = JSON.parse(fs.readFileSync(`${global.appRoot}/assets/server/rest/v1/schemas/billing/billing-setup-payment-method.json`, 'utf8'));
+  private billingInvoicesGet: Schema = JSON.parse(fs.readFileSync(`${global.appRoot}/assets/server/rest/v1/schemas/billing/invoices-get.json`, 'utf8'));
+  private billingInvoiceGetByID: Schema = JSON.parse(fs.readFileSync(`${global.appRoot}/assets/server/rest/v1/schemas/billing/invoice-get-by-id.json`, 'utf8'));
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/validator/BillingValidator.ts:16]: file name is not similar to the others

suggestion:
invoice-get.json

please rename the corresponding file

File: src/server/rest/v1/validator/BillingValidator.ts [hunk 6]
```diff
@@ -39,4 +42,16 @@ export default class BillingValidator extends SchemaValidator {
   public validateBillingSetupUserPaymentMethodReq(data: Record<string, unknown>): HttpSetupPaymentMethod {
     return this.validate(this.billingSetupUserPaymentMethod, data);
   }
+
+  public validateBillingInvoicesGetReq(data: Record<string, unknown>): HttpBillingInvoiceRequest {
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/validator/BillingValidator.ts:46]: Name of the returned type is confusing. 
I would expect an "s"

```suggestion
  public validateBillingInvoicesGetReq(data: Record<string, unknown>): HttpBillingInvoicesRequest {
```

File: src/server/rest/v1/validator/BillingValidator.ts [hunk 7]
```diff
@@ -39,4 +42,16 @@ export default class BillingValidator extends SchemaValidator {
...
+    return this.validate(this.billingInvoicesGet, data);
+  }
+
+  public validateBillingInvoicesGetByIdReq(data: Record<string, unknown>): HttpBillingRequest {
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/validator/BillingValidator.ts:50]: Both the name of the method and name of the returned type are confusing

```suggestion
  public validateBillingInvoicesGetReq(data: Record<string, unknown>): HttpBillingInvoiceRequest {
```

**Review by ClaudeROSSI — CHANGES_REQUESTED** [review]
The billing validator is not similar to the other ones (in term of naming).
Please check my comments and suggestion.

File: src/assets/server/rest/v1/schemas/billing/invoices-get.json (hunk 1 above)
Comment (Ninjeneer) [reply src/assets/server/rest/v1/schemas/billing/invoices-get.json:11]: Both work actually. The difference is that my pattern accept values such as `paid|` . Even if it as no really sense when it's written, this is actually a legit value, what do you think ?

Comment (ClaudeROSSI) [reply src/assets/server/rest/v1/schemas/billing/invoices-get.json:11]: my understanding is that your pattern also accepts something like:
- paidpaiddraft
Does it?

Comment (Ninjeneer) [reply src/assets/server/rest/v1/schemas/billing/invoices-get.json:11]: Oh yes it does, nice catch :) There are missing parenthesis. I'll take yours then

File: src/server/rest/v1/service/BillingService.ts (hunk 2 above)
Comment (Ninjeneer) [reply src/server/rest/v1/service/BillingService.ts]: Renamed

File: src/server/rest/v1/service/BillingService.ts (hunk 3 above)
Comment (Ninjeneer) [reply src/server/rest/v1/service/BillingService.ts]: Renamed

File: src/server/rest/v1/service/security/BillingSecurity.ts (hunk 4 above)
Comment (Ninjeneer) [reply src/server/rest/v1/service/security/BillingSecurity.ts]: Removed

File: src/server/rest/v1/validator/BillingValidator.ts (hunk 5 above)
Comment (Ninjeneer) [reply src/server/rest/v1/validator/BillingValidator.ts]: Renamed

File: src/server/rest/v1/validator/BillingValidator.ts (hunk 6 above)
Comment (Ninjeneer) [reply src/server/rest/v1/validator/BillingValidator.ts]: Renamed

File: src/server/rest/v1/validator/BillingValidator.ts (hunk 7 above)
Comment (Ninjeneer) [reply src/server/rest/v1/validator/BillingValidator.ts]: Renamed

File: src/server/rest/v1/service/BillingService.ts [hunk 8]
```diff
@@ -347,7 +347,7 @@ export default class BillingService {
     UtilsService.assertComponentIsActiveFromToken(req.user, TenantComponents.BILLING,
       Action.LIST, Entity.INVOICE, MODULE_NAME, 'handleGetInvoice');
     // Filter
-    const filteredRequest = BillingSecurity.filterGetInvoiceRequest(req.query);
+    const filteredRequest = BillingValidator.getInstance().validateBillingInvoiceReq(req.query);
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/service/BillingService.ts:350]:
```suggestion
    const filteredRequest = BillingValidator.getInstance().validateBillingInvoiceGetReq(req.query);
```

File: src/server/rest/v1/service/BillingService.ts [hunk 9]
```diff
@@ -618,7 +618,7 @@ export default class BillingService {
     UtilsService.assertComponentIsActiveFromToken(req.user, TenantComponents.BILLING,
       Action.DOWNLOAD, Entity.BILLING, MODULE_NAME, 'handleDownloadInvoice');
     // Filter
-    const filteredRequest = BillingSecurity.filterDownloadInvoiceRequest(req.query);
+    const filteredRequest = BillingValidator.getInstance().validateBillingInvoiceReq(req.query);
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/service/BillingService.ts:621]:
```suggestion
    const filteredRequest = BillingValidator.getInstance().validateBillingInvoiceGetReq(req.query);
```

File: src/server/rest/v1/validator/BillingValidator.ts [hunk 10]
```diff
@@ -39,4 +41,12 @@ export default class BillingValidator extends SchemaValidator {
   public validateBillingSetupUserPaymentMethodReq(data: Record<string, unknown>): HttpSetupPaymentMethod {
     return this.validate(this.billingSetupUserPaymentMethod, data);
   }
+
+  public validateBillingInvoicesGetReq(data: Record<string, unknown>): HttpBillingInvoicesRequest {
+    return this.validate(this.billingInvoicesGet, data);
+  }
+
+  public validateBillingInvoiceReq(data: Record<string, unknown>): HttpBillingInvoiceRequest {
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/validator/BillingValidator.ts:49]:
```suggestion
  public validateBillingInvoiceGetReq(data: Record<string, unknown>): HttpBillingInvoiceRequest {
```

File: postman/EVSE.postman_collection.json [hunk 11]
```diff
@@ -346,45 +346,41 @@
 			"response": []
 		},
 		{
-			"name": "FE-CompanyCreate",
+			"name": "FE-SchneiderGTCGetBorneSud",
```
Comment (ClaudeROSSI) [inline postman/EVSE.postman_collection.json]: @Ninjeneer : Is it normal that we have so many changes in that file?

**Review by ClaudeROSSI — CHANGES_REQUESTED** [review]
still some naming suggestion!
Let me know what you think
Claude

File: postman/EVSE.postman_collection.json (hunk 11 above)
Comment (Ninjeneer) [reply postman/EVSE.postman_collection.json:349]: Yes, I'm also updating the Postman collection along the API migration and the exported JSON is pretty big

File: src/server/rest/v1/service/security/BillingSecurity.ts [hunk 12]
```diff
@@ -39,10 +39,10 @@ export default class BillingSecurity {
     return filteredRequest;
   }
 
-  public static filterGetInvoiceRequest(requestQuery: any): HttpBillingInvoiceRequest {
+  public static filterGetInvoiceRequest(requestQuery: any): HttpBillingInvoicesRequest {
     return {
       ID: sanitize(requestQuery.ID)
-    } as HttpBillingInvoiceRequest;
+    } as HttpBillingInvoicesRequest;
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/service/security/BillingSecurity.ts:45]: Method not used!
Can you please remove old stuff which is not necessary anymore?

**Review by ClaudeROSSI — CHANGES_REQUESTED** [review]
small comment

**Review by ClaudeROSSI — APPROVED** [review]
thanks

**Comment by LucasBrazi06** [comment]
> + postman collection update

You can import the OAS file in Postman in the end.

**Comment by Ninjeneer** [comment]
> You can import the OAS file in Postman in the end.

Didn't know that ! Then, do we remove the JSON collection from the repo ? As anyone can generate its own collection with the OpenAPI file. This could avoid syncing both files and avoid people from using a deprecated collection

**Comment by LucasBrazi06** [comment]
> > You can import the OAS file in Postman in the end.
> 
> Didn't know that ! Then, do we remove the JSON collection from the repo ? As anyone can generate its own collection with the OpenAPI file. This could avoid syncing both files and avoid people from using a deprecated collection

No you can re-generate it once and everyone will use it right away when you will have finished the migration.

**Review by LucasBrazi06 — APPROVED** [review]

**Pull request merged**